extensions/functions_string.yaml

```diff
@@ -1263,6 +1263,74 @@ scalar_functions:
           - value: i32
             name: "count"
         return: "string"
+  -
+    name: string_split
+    description: >-
+      Split a string into a list of strings, based on a specified `separator` character.
+    impls:
+      - args:
+          - value: "varchar<L1>"
+            name: "input"
+            description: The input string.
+          - value: "varchar<L2>"
+            name: "separator"
+            description: A character used for splitting the string.
+        return: "List<varchar<L1>>"
+      - args:
+          - value: "string"
+            name: "input"
+            description: The input string.
+          - value: "string"
+            name: "separator"
+            description: A character used for splitting the string.
+        return: "List<string>"
+  -
+    name: regex_string_split
+    description: >-
+      Split a string into a list of strings, based on a regular expression pattern.  The
```

Comment:
I feel like this could use a bit more explanation. I guess the idea is that it works the same as a regular string split, i.e. removing the substrings matched by the regex from the resulting string list. However, I could also imagine someone interpreting it as the regex picking only the split point, such that every character from the original string ends up in one of the returned list elements. Both implementations would be useful, but the one that removes the matched string is more expressive, because you could wrap the regex in a positive lookahead to mimic the other implementation.

Reply:
Good point.  I updated the description to be more detailed.  From what I've seen, the implementation that removes the matched substring is also how a bunch of different SQL dialects do it.